### DISCUSSION TITLE
Port groups on Dell PowerSwitch devices

### DISCUSSION THREAD

**Discussion by tomkivlin** [netbox-community/netbox#15793] [Ideas]
Dell PowerSwitch (https://www.dell.com/support/manuals/en-sg/dell-emc-smartfabric-os10/smartfabric-os-user-guide-10-5-0/port-groups-on-s5200f-on-switches?guid=guid-1a9f6680-ab7e-4cfa-9c94-96cf3b436f8c&lang=en-us) and maybe others have a concept of port groups, which are fixed into the physical design of the switch. 

Because all interfaces in a port group need to be the same speed, in cases where different speeds are needed on the same switch (e.g. some interfaces SFP+ 10G, some SFP28 25G), planning is needed to ensure only the same speed transceivers & connections are used with interfaces in a given port group. 

Therefore it would be useful to be able to view `port group : interface` mappings to help with the planning of such scenarios, and to validate that this constraint is in the design.

There are sort of a breakout thing, but not using a breakout cable - the "breakout" is within the switch itself.

I looked at modules but I couldn't create the relationship with interfaces. Also tried parent/child interfaces but I couldn't add SFP28 interfaces as a child (came up with the validation error that only virtual interfaces can be child).

I wonder if anyone has a good workaround to manage this scenario?  

Or whether a new "port group" component type that maps to interfaces and enforces this constraint is something to request as a new "nice to have" feature? e.g. to:

1. Add port groups as a component of a device / device type
2. Map / assign interfaces to a port group
3. Validate that all interfaces in a given port group are the same speed
4. View a list of port groups on a device, the interfaces contained within, and what those interfaces are connected to

**Comment by deeconwastaken** [comment]
I am using the same type of switch and unfortunately have not found a reasonable way to model the switch cleanly. The limitations associated with the port groups (all 4 ports can only have the same speed) are also an issue. Are there any ways to implement the port groups as suggested?

![Bild (2)](https://github.com/user-attachments/assets/7ee3ad95-f3ae-4a2a-8126-f0e738310d47)

**Reply by tomkivlin** [reply]
All I've done is to create tags based on the port group names, and apply to each interface.

I did have a dream about using the custom validation to make sure the ports in a port-group all had the same configuration, or a webhook to notify on changes to only one port in a port-group but I've not got around to doing that yet.

**Reply by deeconwastaken** [reply]
I wanted to check if anyone has found a clean way to represent these switches in NetBox (Cisco Nexus 9000 series, Catalyst 9500X/9600X, Arista 7050X, 7280X, 7300X, or some Juniper QFX models).

One challenge I’ve run into is that the interface name changes when the port speed is non-native (e.g. a 10G port on Dell devices gets the suffix :1).

I’ve tried modeling this with virtual interfaces and module bays, but that doesn’t seem to fit well. Using tags is an option, but feels too error-prone in practice.

Has anyone come across a good approach or best practice for handling this? Perhaps someone has even found a solution using a custom script.
